### PR TITLE
Do Not Merge - trigger script to process manifests

### PR DESCRIPTION
**Pull request by laurafitzgerald**
## What
Adding stage to the delorean 2.0 discovery pipeline which triggers a script which will process the manifests out of the image from the UMB message trigger.

### CODE REVIEW

**Comment by mikenairn** [comment]
We should merge https://github.com/integr8ly/ci-cd/pull/414/ before this one, and implement this logic in the "Extract and Process Manifests" stage https://github.com/integr8ly/ci-cd/pull/414/files#diff-9d245d98e5f80fe8a65190c6e62116d5R32-R37

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 1]
```diff
@@ -8,6 +8,20 @@ pipeline {
     }
 
     stages {
+        stage('Process manifest') {
+            steps {
+                cleanWs()
+                script {
+                    final Map msg = readJSON text: CI_MESSAGE
+                    sh 'git clone https://github.com/integr8ly/integreatly-operator.git'
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: Cloning this repo will be done in it's own stage here https://github.com/integr8ly/ci-cd/pull/414/files#diff-9d245d98e5f80fe8a65190c6e62116d5R18-R23

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 2]
```diff
@@ -8,6 +8,20 @@ pipeline {
...
+                    dir("${env.WORKSPACE}/integreatly-operator"){
+                        //This branch can be updated later, just using the pushed one for now
+                        sh 'git checkout 3scale-next'
+                        sh "sh scripts/process-image-manifests ${msg.info.extra.image.index.pull[1]}"
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: no need for the "sh" here

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 3]
```diff
@@ -8,6 +8,20 @@ pipeline {
...
     stages {
+        stage('Process manifest') {
+            steps {
+                cleanWs()
+                script {
+                    final Map msg = readJSON text: CI_MESSAGE
+                    sh 'git clone https://github.com/integr8ly/integreatly-operator.git'
+                    dir("${env.WORKSPACE}/integreatly-operator"){
+                        //This branch can be updated later, just using the pushed one for now
+                        sh 'git checkout 3scale-next'
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: This logic will be dealt with in it's own stage here https://github.com/integr8ly/ci-cd/pull/414/files#diff-9d245d98e5f80fe8a65190c6e62116d5R25-R30

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 4]
```diff
@@ -1,38 +1,49 @@
...
     }
-
     stages {
 
         stage('Process UMB Message') {
             steps {
-                //ToDo Process the input UMB Message (CI_MESSAGE) to extract information required by the following stages
-                echo "Process UMB Message"
+                script{
+                    final Map msg = readJSON text: CI_MESSAGE
```
Comment (laurafitzgerald) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: There's something wrong with how I'm trying to do this. Will revisit it later. https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/Delorean%202.0/job/delorean-2-ews/job/3scale/job/discovery/18/console

**Comment by laurafitzgerald** [comment]
@mikenairn @StevenTobin Feel free to close this one if it becomes out of date. :)

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
-    agent none
+    agent any
 
     parameters {
         // https://datagrepper.engineering.redhat.com/id?id=ID:messaging-devops-broker02.web.prod.ext.phx2.redhat.com-44775-1580379079663-8:893046:0:0:1&is_raw=true&size=extra-large
         string defaultValue: '', description: 'This will be populated by the CI trigger', name: 'CI_MESSAGE', trim: true
     }
-
     stages {
+        stage('Install yq') {
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: Maybe make this stage more generic `stage('Ensure Dependencies')`, we will likely remove the yq installation, but i guess we could have a stage to check stuff before continuing.

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 6]
```diff
@@ -17,22 +26,31 @@ pipeline {
...
+                    } 
+                }
             }
         }
 
         stage('Extract and Process Manifests') {
             steps {
-                //ToDo Extract the manifests from a container image and process them as required
-                echo "Extract and Process Manifests"
+                dir("${env.WORKSPACE}/integreatly-operator"){
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: Any reason you need ${env.WORKSPACE}, you should already be in that directory, I'd probably also just do a dir round the whole lot after the checkout tbh, not sure how that works in the declarative pipelines though.

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 7]
```diff
@@ -17,22 +26,31 @@ pipeline {
...
+                    // final Map msg = readJSON text: CI_MESSAGE
+                    sh "scripts/process-image-manifests registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-3scale-rhel7-operator-metadata:1.11.0-3 3scale 3scale-operator"
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: I assume this isn't meant to be hard coded

**Comment by laurafitzgerald** [comment]
@mikenairn Thanks for the review. I don't think we'll actually merge this one cos @StevenTobin is working on this stuff. I haven't been paying much attention to making the code here good. I'm just using it to test the script and pushing it in case there might be something useful to take into other prs which @StevenTobin is working on.

File: jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile [hunk 8]
```diff
@@ -17,22 +16,30 @@ pipeline {
 
         stage('Clone Operator Repo') {
             steps {
-                //ToDo Clone the integeratly-operator repo
-                echo "Clone Operator Repo"
+                cleanWs()
```
Comment (StevenTobin) [inline jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile]: This should prob be higher up actually so i'm not wiping anything that happens earlier

**Comment by laurafitzgerald** [comment]
Closing in favour of https://github.com/integr8ly/ci-cd/pull/423

**Pull request closed without merging**